include/iio/iio.h

```diff
@@ -445,6 +445,27 @@ iio_attr_get_filename(const struct iio_attr *attr);
 __api __pure const char *
 iio_attr_get_static_value(const struct iio_attr *attr);
 
+/** @brief Extract the range from a string originating from an '_available" attribute
+ * @param input A string that is expected to have the [min step max] format
+ * @param min Will be set with the range minimum value if string was successfully parsed
+ * @param step Will be set with the range step value if string was successfully parsed
+ * @param max Will be set with the range maximum value if string was successfully parsed
+ * @return On success, 0 is returned
+ * @return On error, a negative errno code is returned */
+__api __pure
+int iio_get_range_from_available(const char *input, double *min, double *step, double *max);
+
```

Comment:
I'm not sure why the application needs to read the channel _attribute_available attribute, and then pass it here, rather than just passing the channel_attribute to this in the first place? (like the way that the rest of libiio works?)

Reply:
Not all attributes have the "_available" which made the think of avoiding misusage. One would need to actually read the attribute and then pass the data further to this helper. But I have thought about inconsistency with the rest of the API). Also, this approach causes the buffer storing the attribute value to be create twice: once when attribute gets read and once inside the helper.
It makes sense to change the signature so that it takes an `iio_attr` as parameter.